### PR TITLE
HL-97

### PR DESCRIPTION
**Pull request by xiangzhe0812**
[Link to ticket](https://dltx.atlassian.net/jira/software/projects/HL/boards/82?selectedIssue=HL-97)

Screenshot 2022-11-17 at 4 35 50 pm
Screenshot 2022-11-17 at 4 35 44 pm

### CODE REVIEW

File: src/pages/HorseRace/components/Back/Back_Logic.tsx [hunk 1]
```diff
@@ -205,6 +208,16 @@ const BackLogic: React.FC<Props> = ({ runner }) => {
...
   const [wagerAmount, setWagerAmount] = useState<number>(0);
   const [signature, setSignature] = useState<EcSignature>();
+  const config = useConfig();
+
+  const marketData = useMarketDetail(selectedMarketAddress);
+  const { data: balanceData } = useBalance({
+    address: address as Address,
+    token:
+      marketData?.name && marketData.name.includes("DAI")
+        ? (config?.tokenAddresses.DAI as Address)
```
Comment (brettvniekerk) [inline src/pages/HorseRace/components/Back/Back_Logic.tsx:218]: these can be imported from `StaticConfig`

Comment (brettvniekerk) [reply src/pages/HorseRace/components/Back/Back_Logic.tsx:218]: also is `as Address` necessary? is there not an `Address` constructor?

File: src/pages/HorseRace/components/Back/Back_View.tsx [hunk 2]
```diff
@@ -25,6 +25,7 @@ type Props = {
   };
   isEnoughAllowance: boolean;
   handleBackContractWrite: () => Promise<void>;
+  balanceData?: any;
```
Comment (brettvniekerk) [inline src/pages/HorseRace/components/Back/Back_View.tsx:28]: why `any`? is there not a type?

File: src/pages/HorseRace/components/Back/Back_View.tsx [hunk 3]
```diff
@@ -75,10 +77,22 @@ const BackView: React.FC<Props> = ({
               className="w-full rounded-md border-gray-300 shadow-sm focus:border-indigo-300 focus:ring focus:ring-indigo-200 focus:ring-opacity-50 mb-2"
             />
           </label>
-
-          <span className="py-1">
-            Payout: <span>{potentialPayout}</span>
-          </span>
+          <div className="grid grid-cols-2 gap-4">
+            <div>
```
Comment (brettvniekerk) [inline src/pages/HorseRace/components/Back/Back_View.tsx:81]: no need for empty grid tag - u can add `block` to the `<span>` classname if you want it to behave like a div

File: src/pages/HorseRace/components/Back/Back_View.tsx [hunk 4]
```diff
@@ -75,10 +77,22 @@ const BackView: React.FC<Props> = ({
...
+              <span className="py-1">
+                Payout: <span>{potentialPayout}</span>
+              </span>
+            </div>
+            <div className="text-right text-red-600">
+              <span className="py-1">
+                Available:{" "}
+                <span>
+                  {balanceData ? balanceData.formatted : "0"}{" "}
```
Comment (brettvniekerk) [inline src/pages/HorseRace/components/Back/Back_View.tsx:90]: im assuming `balanceData` is populated if the user has their wallet connected - do we not want to handle an unconnected wallet here?

File: src/pages/HorseRace/components/Back/Back_Logic.tsx (hunk 1 above)
Comment (xiangzhe0812) [reply src/pages/HorseRace/components/Back/Back_Logic.tsx:218]: 1. no, what in staticConfig are mock asset address, what i got from dropdown is market contract address, 
so it has to be this way.
2. you are right.

File: src/pages/HorseRace/components/Back/Back_View.tsx (hunk 2 above)
Comment (xiangzhe0812) [reply src/pages/HorseRace/components/Back/Back_View.tsx:28]: no type exported from wagmi

File: src/pages/HorseRace/components/Back/Back_View.tsx (hunk 4 above)
Comment (xiangzhe0812) [reply src/pages/HorseRace/components/Back/Back_View.tsx:90]: which it will be 0.

**Review by pondpiu — DISMISSED** [review]

File: src/pages/HorseRace/components/Back/Back_Logic.tsx [hunk 5]
```diff
@@ -205,6 +208,16 @@ const BackLogic: React.FC<Props> = ({ runner }) => {
...
   const [wagerAmount, setWagerAmount] = useState<number>(0);
   const [signature, setSignature] = useState<EcSignature>();
+  const config = useConfig();
+
+  const marketData = useMarketDetail(selectedMarketAddress);
+  const { data: balanceData } = useBalance({
+    address: address,
+    token:
+      marketData?.name && marketData.name.includes("DAI")
+        ? (config?.tokenAddresses.DAI as Address)
```
Comment (brettvniekerk) [inline src/pages/HorseRace/components/Back/Back_Logic.tsx:218]: import `StaticConfig` for this so you dont need to use the optional flag

**Review by pondpiu — APPROVED** [review]

**Review by bitcoinbrisbane — APPROVED** [review]

**Review by brettvniekerk — APPROVED** [review]

**Pull request merged**